### ISSUE TITLE
Version 1.8.2 three times slower to launch

### ISSUE THREAD

**Issue by technoblogy** [arduino/Arduino#6350]
I've been comparing the time to launch 1.6.8, 1.8.1, and 1.8.2 on a MacBook Pro (Mac OS X 10.11.6).

Both 1.6.8 and 1.8.1 take 4 seconds, but 1.8.2 takes 13 seconds, over three times slower.

Any suggestions why this might be?

**Comment by PaulStoffregen** [comment]
Maybe the newer JRE?

**Comment by gilbertoverg** [comment]
I have the same issue, for me the launch time for 1.8.2 and 1.8.3 is 27 seconds

**Comment by bperrybap** [comment]
Some of the problem (maybe most of it) is due to something in the IDE that ends up scanning the users sketchbook libraries and cores multiple times. It does this on startup or any time you switch cores.
I don't understand multiple scans would ever be necessary.

This has existed for many years but seems to have gotten worse recently.
It gets particularly slow when your sketchbook directory is across a network mount.

In the past it was very obvious for me as certain IDEs had some foolish naming restrictions on library directory names and I would have to answer the same prompt for a given library directory name up to 9 times for each directory it didn't like on startup or whenever I switched cores.
i.e. I'd get a slew warnings for directory names it didn't like - each of which must be individually acknowledged, and then this same sequence would repeat over and over again until the IDE finally started. 
It was a total pain.
The new IDEs no longer have the restrictions on directory names and hence don't toss up the warnings that you must acknowledge but they seem to still do the multiple scans of the users sketchbook libraries.

**Comment by gilbertoverg** [comment]
@bperrybap I agree with you, I can see Arduino scanning libraries during startup with:
```sudo fs_usage pid <Arduino pid>```

**Comment by technoblogy** [comment]
Thank you for confirming that I'm not the only one who has found this. I definitely think it would be worth the developers spending a bit of time to fix this.

**Comment by cmaglie** [comment]
The multiple scanning of the libraries can be surely optimized, but it has been like this well before 1.8.1, IMHO the cause of the slow down should be searched elsewhere.

**Comment by bengtmartensson** [comment]
I have found that "phony" serial ports (e.g. Bluetooth virtual such) can be startup time culprit. Can this be an issue here?

**Comment by f3rland** [comment]
Thanks for pointing that out @bengtmartensson 
I had multiple "VirtualBox Host-Only Ethernet Adapter #x" created by Virtual box/Vagrant and Arduino IDE (1.8.2) took nearly 1 minute to load before I removed all of those adapter. Now 3 seconds top.

**Comment by gilbertoverg** [comment]
I've tried the test build from #6419 but it still takes 35 seconds to launch

**Comment by apphead** [comment]
Issue still present in 1.8.4

**Comment by jaygarcia** [comment]
Want to confirm that this is still present in 1.8.5. =)

**Comment by facchinm** [comment]
@jaygarcia could you test 1.9.0 beta? It should contain an additional fix (5c61508) that should solve the issue. Thansk!

**Comment by gilbertoverg** [comment]
1.9.0 beta solves the issue for me

**Comment by technoblogy** [comment]
I confirm that 1.9.0 beta launches in 4 seconds, matching the behaviour of 1.6.8 and 1.8.1. That's great news - a big thank you to everyone who worked to achieve this!

**Comment by SuperTango** [comment]
@jaygarcia dunno if this still matters, but I see this problem in 1.8.5.

**Comment by jaygarcia** [comment]
@SuperTango this seems to be fixed in 1.9 beta: https://www.arduino.cc/en/Main/Software#beta

This ticket can most likely be closed once 1.9 is released to prod.

Fantastic work everyone and thank you so much for fixing this! <3
